sphinx/util/inspect.py

```diff
@@ -372,6 +444,14 @@ def __init__(self, subject: Callable, bound_method: bool = False,
             # we try to build annotations from argspec.
             self.annotations = {}
 
+        # merge type_comment-based type hints
```

Comment:
Note: I just added a code to here. But it would be better to add a hook point outside of `Signature` class. It would be also needed a handler for stub files (.pyi). It would be better to enhance online object itself via `obj.__annotations__` interface.

Reply:
see #4824

Reply:
Done.